### ISSUE TITLE
Support pathLength via CSS

### ISSUE THREAD

**Issue by argyleink** [w3c/svgwg#773]
I'd expected to be able to set this attribute via CSS but turns out that I can't. It was also pointed out here in this tutorial on an svg animation by @shshaw https://youtu.be/7aobRPg7BXI?t=741

**Comment by shshaw** [comment]
I would definitely love to see this supported. Being able to set pathLength with CSS would make stroke effects & animations even easier. I can imagine some great effects by animating pathLength as well: https://codepen.io/shshaw/pen/KKwRMZd?editors=1010

**Comment by tabatkins** [comment]
Hm, I'm surprised `path-length` isn't a geometry property for it, yeah.

**Comment by AmeliaBR** [comment]
I agree that it makes sense to treat `pathLength` as another geometry property. And I also agree that the logical property name would be `path-length` (so that the camel-cased DOM version is still `style.pathLength`.

So the question is: who's interested in implementing it? What spec version should this be targetted for?

**Comment by fsoder** [comment]
Given sufficiently positive signals from other engine vendors, I'd be happy to implement (or aid an implementation of) the property (and presentation attribute mapping) in Blink.

**Comment by grorg** [comment]
Consider this a positive signal from WebKit!

**Comment by Tavmjong** [comment]
I've done some basic testing. The attribute 'pathLength' is supported by Firefox and Chrome, not by Safari and Edge. Inkscape does not support 'pathLength' but it shouldn't be hard to implement.

The attached zip file contains tests of stroke dashing using 'pathLength':
* path_length.svg: Tests basic support on path, rectangle, circle.
* path_length2.svg: Tests support on a very long spiral path consisting of cubic Beziers. Firefox and Chrome differ in the calculation of the path length by about 0.1% (14px out of 11760px). They both agree within 0.2% of my naive calculation of the path length.
* path_length3.svg: A SMIL animation on the spiral path, with and without using 'pathLength'. There is no indication of a problem with computation time.

I see no problems in adding 'path-length' as a CSS property.

[path_length.zip](https://github.com/w3c/svgwg/files/4108982/path_length.zip)

**Comment by fsoder** [comment]
Chromium/Blink bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1045430

I looked at what tests existed for the 'pathLength':

https://wpt.fyi/results/svg?label=master&label=experimental&aligned&q=pathlength

That seemed to cover the visual tests, and there are a few related to SVGGeometryElement APIs in types/scripted/.

I'm in the process of exporting some of Blink's tests as well (ought show up in the above query when the export has completed). And I see that (while I was writing this =P) @Tavmjong has drafted a few more test.

**Comment by argyleink** [comment]
🙂 ping! curious if this has moved forward!?

**Comment by longsonr** [comment]
Currently the path for a textPath can be display: none and it's OK. If you make pathLength a CSS property then display:none paths used for textPath that have pathLength set will no longer work as expected.

**Comment by viralipurbey** [comment]
> Chrome is currently being implemented: https://chromium-review.googlesource.com/c/chromium/src/+/7571603

Updating the CL link: https://chromium-review.googlesource.com/c/chromium/src/+/7607639. Sorry for the confusion @yisibl

**Comment by dirkschulze** [comment]
RESOLUTION: pathLength attribute will get a presentation attribute with the CSS property being path-length taking a number unit only for now
https://www.w3.org/2026/03/26-svg-minutes.html

**Comment by chrishtr** [comment]
I think the CSSWG will also need to review this property and potentially change its syntax to align with unit type conventions in CSS. [See comment here](https://github.com/WebKit/standards-positions/issues/621#issuecomment-4401672694).

**Comment by viralipurbey** [comment]
Adding the resolution to the right issue.
RESOLUTION: Follow the suggestion of the CSS WG and let pathLength take a length value rather than a number value.

https://www.w3.org/2026/05/21-svg-minutes.html
